plugins/flatpak/gs-flatpak.c

```diff
@@ -1301,6 +1320,77 @@ gs_flatpak_app_install_source (GsFlatpak *self, GsApp *app,
 	return TRUE;
 }
 
+static GsApp *
+get_main_app_of_related (GsFlatpak *self,
+			 GsApp *related_app,
+			 GCancellable *cancellable,
+			 GError **error)
+{
+	g_autofree gchar *main_app_name = g_strdup (gs_flatpak_app_get_ref_name (related_app));
+	g_autoptr(FlatpakInstalledRef) ref = NULL;
+	const gchar *app_name;
+	g_auto(GStrv) app_tokens = NULL;
+
+	app_name = gs_flatpak_app_get_main_app_ref (related_app);
+	if (app_name == NULL) {
+		g_set_error (error, G_IO_ERROR, G_IO_ERROR_NOT_FOUND,
+			     "%s doesn't have a main app set to it.",
+			     gs_app_get_unique_id (related_app));
+		return NULL;
+	}
+
+	app_tokens = g_strsplit (app_name, "/", -1);
+	if (g_strv_length (app_tokens) != 4) {
+		g_set_error (error, G_IO_ERROR, G_IO_ERROR_INVALID_DATA,
+			     "The main app of %s has an invalid name: %s",
+			     gs_app_get_unique_id (related_app), app_name);
+		return NULL;
+	}
+
+	ref = flatpak_installation_get_installed_ref (self->installation,
+						      FLATPAK_REF_KIND_APP,
+						      app_tokens[1],
+						      app_tokens[2],
+						      app_tokens[3],
+						      cancellable,
+						      error);
+
+	if (ref == NULL)
+		return NULL;
+
+	return gs_flatpak_create_installed (self, ref, error);
+}
+
+static GsApp *
+get_real_app_for_update (GsFlatpak *self, GsApp *app, GCancellable *cancellable,
+			GError **error)
```

Comment:
Wrong indentation

Reply:
Also, seeing other functions (e.g. `get_main_app_of_related()`)  it feels like you want to put each param in a single line

Reply:
Hmm... after reading the rest of the code and realizing that `get_real_app_for_update()` is not expected to ever return NULL (thus it's not expected to fail), it looks like you should not pass a `GCancellable *cancellable` here, and simply use `NULL` below when calling `get_main_app_of_related()`, since there's nothing you would do in case the operation gets cancelled.

But that depends on whether it's realistically possible to get this action cancelled or not:
  * If it's possible to cancel this function, I think you should keep both the `GCancellable` and the `GError **` parameters, handle the cancellable error as a real error below and therefore accepting there's at least a case where this function can return NULL, thus checking for that option in the caller.
  * If it's NOT possible to cancel this function, then you should simply drop both the `GCancellable` and the `GError **` parameters from this function, and simply use `NULL` below when calling `get_main_app_of_related()`, as mentioned above (and add an assertion in the caller, after using this function).

Reply:
About adding a parameter per line, it's not really something enforced or recommended in g-s, but I will change it.

About the function never returning NULL, you're right. It's a bit messy and I think that we should instead abort when the error is other than `NOT_FOUND`.